src/plans/components/ProviderSelect.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ComponentProps, FC, ReactNode } from 'react';
+import { getResourceUrl } from 'src/modules/Providers';
+
+import { ExternalLink } from '@components/common/ExternalLink';
+import Select from '@components/common/Select';
+import { ProviderModelGroupVersionKind, ProviderModelRef, V1beta1Provider } from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { EmptyState, EmptyStateVariant, SelectOption } from '@patternfly/react-core';
+import { ForkliftTrans, useForkliftTranslation } from '@utils/i18n';
+
+type ProviderSelectProps = Pick<ComponentProps<typeof Select>, 'onSelect' | 'status'> & {
+  id: string;
+  value: string;
+  namespace: string;
+  placeholder?: string;
+  emptyState?: ReactNode;
+};
+
+export const ProviderSelect: FC<ProviderSelectProps> = ({
+  id,
+  value,
+  namespace,
+  placeholder,
+  emptyState,
+  status,
+  onSelect,
+}) => {
+  const { t } = useForkliftTranslation();
+  const [providers, providersLoaded, providerError] = useK8sWatchResource<V1beta1Provider[]>({
+    namespace,
+    groupVersionKind: ProviderModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+  });
+
+  const providersListUrl = getResourceUrl({
+    reference: ProviderModelRef,
+    namespace: namespace,
+  });
+
+  return (
+    <Select
+      id={id}
+      value={value}
+      status={status}
+      onSelect={onSelect}
+      placeholder={
+        placeholder ||
+        (!providersLoaded || providerError ? t('Providers failed to load.') : undefined)
```

Comment:
I don't understand this condition. Why not display the error from the server? 
Why is it a placeholder?

Reply:
Basically, there's a situation where the field isn't touched, so form errors aren't triggered, yet we could let the user know still that there aren't any results to show for the dropdown, and that's where the placeholder comes into play.

We could just display the server error, but I'm not sure how readable that would be, so that's why I went with something not very verbose and readable within the input.

Considering these things, I could just trust the providerError's message is good enough and not have our own here, but if we don't want the message in the placeholder at all, then the fetch of providers might need to be moved outside of this component. What are your expectations here?

Reply:
As a rule of thumb, UI should present server errors and not try to map or override them.
I would present an alert or something in case of server error, not in a placeholder. 

I think a loader should be presented until all wizard resources are fetched, and if a mandatory resource is missing, the error should be presented instead of the form.

Reply:
Instead of the whole form or just the 1 field in the form? I think this could use some design attention. Right now, I see no harm in failing silently, where we don't have to have a placeholder/disabled state if no providers get fetched, we can just treat it the same way as if the fetch was successful and the array of values was empty.

Reply:
I spoke with @mmenestr about this and we came up with the following to present to users when there are API errors for Selects, so I've updated accordingly. We can tweak this more later to allow for refresh or different copy.

<img width="740" alt="image" src="https://github.com/user-attachments/assets/c1c22558-97d8-4ea7-b0ec-c34d39a976ac" />